lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class ReplicaSet < KubernetesResource
+    TIMEOUT = 5.minutes
+
+    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
+      @name = name
+      @namespace = namespace
+      @context = context
+      @file = file
+      @parent = parent
+      @logger = logger
+      @pods = []
+      @rollout_data = { "replicas" => 0 }
```

Comment:
The reason has to do with the possible origins of these resource types:
- Deployments _always_ come from a file in your deploy directory and get their observed state via `sync`.
- ReplicaSets _usually_ show up via deployments, but it is possible to deploy them separately, so I've supported that here (and added a test).
- Pods are very commonly created both ways: unmanaged from a file in your deploy directory, or via a deployment/replicaSet.

For deployments, we have it easy: we see the file, create the instance, and then call sync on it repeatedly to reset and update the observed state. For the other two, there is the additional possibility that they are being created when their parent syncs, and that the parent already has the data on their observed state. In that case, we create the instance and then call `interpret_json_data` with that data; `sync` is never called. Effectively, `interpret_json_data` is the common core between the two flows. I'll investigate making it private / clarifying these two flows.